src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEReader.cs

```diff
@@ -526,7 +527,7 @@ public PEMemoryBlock GetSectionData(string sectionName)
         /// <exception cref="InvalidOperationException">PE image not available.</exception>
         public ImmutableArray<DebugDirectoryEntry> ReadDebugDirectory()
         {
-            var debugDirectory = PEHeaders.PEHeader.DebugTableDirectory;
+            var debugDirectory = PEHeaders.PEHeader!.DebugTableDirectory;
```

Comment:
might be better assert

Reply:
Why, out of curiosity?

Reply:
For me because it is public method could have called without proper initialization and by asserting that could have catched on development phase